clumper/clump.py

```diff
@@ -174,19 +185,112 @@ def read_csv(cls, path, delimiter=",", fieldnames=None, n=None):
+        # python's csv module treats null values as empty strings when writing to a csv -
+        # https://docs.python.org/3.8/library/csv.html#csv.DictWriter
+        # the user can choose to explicitly show missing keys/null values in the dictionary
+        # by assigning `ignore` to the na_values argument. At the moment, the default for
+        # null values are empty string ("") and "NA".
+
+        if na_values == "ignore":
+            result = result
+        else:
+            na_values = ["", "NA"]
+            result = [
+                {key: value for key, value in entry.items() if value not in na_values}
+                for entry in result
+            ]
+
+        # The csv module reads every row as a string, there are no data type assumptions.
+        # This function attempts to solve this. The user can pass a string of either
+        # ('int', 'str', 'float') or if the user knows the keys/fieldnames, can pass a
+        # dictionary mapping the key to the data type.
+        # Technically 'str' data type is not needed, since data is read in as strings anyway.
+
+        if not (isinstance(dtype, (dict, str)) or dtype is None):
+            raise TypeError(
+                """dtype should be a dictionary pair of key and data type, or a single string data type"""
+            )
+
+        dtype_mapping = {"int": int, "float": float, "str": str}
+        if dtype:
+            if isinstance(dtype, str) and dtype in ("int", "float", "str"):
+                result = [
+                    {key: dtype_mapping[dtype](value) for key, value in entry.items()}
+                    for entry in result
+                ]
+            else:
+                result = [
+                    {
+                        key: dtype_mapping[dtype[key]](value) if key in dtype else value
+                        for key, value in entry.items()
+                    }
+                    for entry in result
+                ]
+
+        return Clumper(result)
+
+    def write_csv(self, path, mode="w"):
+        """
+        Write to a csv file.
+
+        Arguments:
+        path: filename
+        mode: `w` writes to a file if it does not exist, or overwrites if it already exists,
+               while `a`: - append to file if it already exists. The default is `w`.
+
+        Note that null values will be exported as empty strings; this is the convention chosen by Python.
+
+        Usage:
+
+        ```python
+        from clumper import Clumper
+
+        from tempfile import TemporaryDirectory
+        from pathlib import Path
+        with TemporaryDirectory() as temporary_dir:
+            Path(temporary_dir).touch('monopoly_copy.csv')
+            path = Path(temporary_dir)/'monopoly.csv'
+            Clumper.read_csv("tests/data/monopoly.csv").write_csv(str(path))
+            reader = Clumper.read_csv(str(path))
+            assert Clumper.read_csv("tests/data/monopoly.csv").collect() == reader.collect()
```

Comment:
Since this is a docstring example it should suffice to just add `/tmp/monopoly.csv` as a filepath.